patches/chrome-browser-profiles-profile.h.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/profiles/profile.h b/chrome/browser/profiles/profile.h
+index 92baa51c809448a43b09ecd7741654a3920cd931..79247ee29828df2ff80ee7dad6b6002e0ebff3f3 100644
+--- a/chrome/browser/profiles/profile.h
++++ b/chrome/browser/profiles/profile.h
+@@ -163,6 +163,7 @@ class Profile : public content::BrowserContext {
+     std::string Serialize() const;
+ #endif
+ 
++   BRAVE_PROFILE_OTR_PROFILE_ID_H
```

Comment:
so I think we can do this in `chromium_src/chrome/browser/profiles/profile.h` without adding additional patch
```
#define profile_id_ \
  profile_id_; \
  friend class TorProfileManager;
```

Reply:
@darkdh I don't think that work because that would cause some replacements in this other part of profile.h:
```c++
class OTRProfileID {
   public:

   [...]

    bool operator==(const OTRProfileID& other) const {
      return profile_id_ == other.profile_id_;
    }

    bool operator!=(const OTRProfileID& other) const {
      return profile_id_ != other.profile_id_;
    }

    bool operator<(const OTRProfileID& other) const {
      return profile_id_ < other.profile_id_;
    }

   [...]

    const std::string profile_id_;
  };
```

Reply:
right, I didn't notice those.

Reply:
What if we just squeeze it into the `PrimaryID` override that we already have:

#define PrimaryID PrimaryID(); **friend class TorProfileManager;** static const OTRProfileID TorID

Reply:
yeah, that works and no patch needed

Reply:
I thought of that but I was trying to keep the friendship in the private section, which is why I didn't propose this before. On a second thought, it seems that avoiding patching is more beneficial than keeping the friendship in the private section, so I've changed it.

PTAL, thanks!